modules/compiler/CMakeLists.txt

```diff
@@ -46,9 +46,9 @@ if(CA_RUNTIME_COMPILER_ENABLED)
     ${CMAKE_CURRENT_SOURCE_DIR}/library/source/library.cpp
     ${COMPILER_CONFIG_SOURCE})
   target_include_directories(compiler-static PUBLIC
-    ${CMAKE_CURRENT_SOURCE_DIR}/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/utils/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/library/include
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/utils/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/library/include>
```

Comment:
Does it work if we do `$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include ${CMAKE_CURRENT_SOURCE_DIR}/utils/include ${CMAKE_CURRENT_SOURCE_DIR}/library/include>` rather than repeating `$<BUILD_INTERFACE:` every time? (Likewise for other `CMakeLists.txt` files.) If that doesn't work, then this looks fine to me as it is.

Reply:
it does seem to work.

Reply:
actually it doesn't seem to work fully. I think I'd rather go with what we have,.